validation/elliptic_solvers/doubly_bounded_poisson.jl

```diff
@@ -0,0 +1,159 @@
+using Oceananigans
+using Oceananigans.Architectures: child_architecture, device
+using Oceananigans.Operators: ∇²ᶜᶜᶜ
+using KernelAbstractions: @kernel, @index, Event
+using Oceananigans.Utils: launch!
+using Oceananigans.BoundaryConditions: fill_halo_regions!
+using Oceananigans.Solvers: FFTBasedPoissonSolver, solve!, PreconditionedConjugateGradientSolver, MultigridSolver
+using Oceananigans.Architectures: architecture, arch_array
+using IterativeSolvers
+using Statistics: mean
+using AlgebraicMultigrid: RugeStubenAMG
+using OffsetArrays
+using GLMakie
+
+import Oceananigans.Solvers: precondition!
+
```

Comment:
I've been wondering for a while if we should include a brief description of what the code does at the start of the every validation script. If people agree, maybe we could start with this one?

Reply:
Sounds good, I'm not very familar with the other validation scripts but would this be a reasonable summary?

"Solves the 2D doubly bounded Poisson equation with the FFT Solver, the Conjugate Gradient Solver, the Multigrid Solver and the Conjugate Gradient Solver preconditioned with the Multigrid Solver."

Reply:
That's good! The only thing I would ask is: does it solve the Poisson equation for what problem? Also, I'd recommend including a similar summary for the other validation script you added if it's not too much trouble! :)

Reply:
It’s for a “fictious” problem. We devise a top-hat source term at the top. So perhaps just write “Demonstrates how one can solve a Poisson problem using the X, Y, Z solver”

Reply:
Sounds good, also with regard to the other validation script that was to demonstrate a yet to be resolved concern so probably shouldn't be merged, thank you for raising that! I'm not sure how best to deal with this

Reply:
Ah, I see. Thanks!

Reply:
Perhaps remove it for now. We can open an issue + new PR about the preconditioning after we merge this one.